### PR TITLE
Add junos L3 interface resource model

### PR DESCRIPTION
**Pull request by danielmellado**

### CODE REVIEW

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 441cee9d9ef1ac3585ef950c26a1c7f78aed9113

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 263d68347fa6d712777b2677689427c664449251

File: models/junos/l3_interfaces/overridden_example_01.txt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    config:
+      - name: ge-0/0/1
+        ipv4:
+           - address: 192.168.1.10/24
+        ipv6:
+           - address: 8d8d:8d01::1/64
+      - name: ge-0/0/2
+        ipv6:
+           - address: 2001:db8:3000::/64
+    state: overridden
```
Comment (ganeshrn) [inline models/junos/l3_interfaces/overridden_example_01.txt]: In the case of overridden if other interfaces have l3 related configuration apart from ge-0/0/1 and ge-0/0/2 it should be deleted in this example. Probably one more interface say ge-0/0/3 can be added in before config and show the l3 config is deleted from ge-0/0/3 in the after state.

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 6223e4d82e9200d8ba351cc04ffc346a6dc22461

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 7ce9d873ffaa8faae26ab890bf9af07fe5812da9

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 5485ac4b4e3a48cc6428eae8d267c96ddd4722fc

File: models/junos/l3_interfaces/junos_l3_interfaces.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+NETWORK_OS: junos
+RESOURCE: l3_interfaces
+COPYRIGHT: Copyright 2019 Red Hat
+LICENSE: gpl-3.0.txt
+
+DOCUMENTATION: |
+  module: junos_l3_interfaces
+  version_added: 2.9
+  short_description: Manage Layer-3 interface on Juniper JUNOS devices
```
Comment (samccann) [inline models/junos/l3_interfaces/junos_l3_interfaces.yml]:
```suggestion
  short_description: Manage Layer 3 interface on Juniper JUNOS devices
```

File: models/junos/l3_interfaces/junos_l3_interfaces.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  description: This module provides declarative management of a Layer-3 interface on Juniper JUNOS devices
```
Comment (samccann) [inline models/junos/l3_interfaces/junos_l3_interfaces.yml]:
```suggestion
  description: This module provides declarative management of a Layer 3 interface on Juniper JUNOS devices
```

File: models/junos/l3_interfaces/junos_l3_interfaces.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  author: Daniel Mellado (@dmellado)
+  notes:
+    - 'Tested against JunOS v18.4R1'
+  options:
+    config:
+      description: A dictionary of Layer-3 interface options
```
Comment (samccann) [inline models/junos/l3_interfaces/junos_l3_interfaces.yml]:
```suggestion
      description: A dictionary of Layer 3 interface options
```

File: models/junos/l3_interfaces/junos_l3_interfaces.yml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+          type: str
+          required: True
+        unit:
+          description:
+            - Logical interface number. Value of C(unit) should be of type integer
+          default: 0
+        ipv4:
+          description:
+            - IPv4 addresses to be set for the Layer 3 logical interface mentioned in I(name) option.
+              The address format is <ipv4 address>/<mask>, the mask is number in range 0-32
```
Comment (samccann) [inline models/junos/l3_interfaces/junos_l3_interfaces.yml]:
```suggestion
              The address format is <ipv4 address>/<mask>. The mask is number in range 0-32
```

File: models/junos/l3_interfaces/junos_l3_interfaces.yml [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+              for example, 192.0.2.1/24, or C(dhcp) to query DHCP for an IP address
```
Comment (samccann) [inline models/junos/l3_interfaces/junos_l3_interfaces.yml]:
```suggestion
              for example, 192.0.2.1/24, or C(dhcp) to query DHCP for an IP address.
```

Comment (samccann) [reply models/junos/l3_interfaces/junos_l3_interfaces.yml]: Should this description be here or in the IPv4 suboption below?  see https://github.com/ansible-network/resource_module_models/pull/19/files#diff-1d5185730444116efbcb1fbcf059cc30R34 for how eos handles this

File: models/junos/l3_interfaces/junos_l3_interfaces.yml [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+          suboptions:
+            address:
+              description:
+                - IPv4 address to be set for the specific interface
+              type: str
+        ipv6:
+          description:
+            - IPv6 addresses to be set for the Layer 3 logical interface mentioned in I(name) option.
+              The address format is <ipv6 address>/<mask>, the mask is number in range 0-128
+              for example, 2001:db8:2201:1::1/64 or C(auto-config) to use SLAAC
```
Comment (samccann) [inline models/junos/l3_interfaces/junos_l3_interfaces.yml]: same comment about moving the detailed description to the suboption below.

File: models/junos/l3_interfaces/junos_l3_interfaces.yml (hunk 6 above)
Comment (danielmellado) [reply models/junos/l3_interfaces/junos_l3_interfaces.yml]: Well, I think it comes clearer there, but I don't have a strong opinion on that, will update the PR but I can just move there if you think it's better.

File: models/junos/l3_interfaces/junos_l3_interfaces.yml (hunk 7 above)
Comment (danielmellado) [reply models/junos/l3_interfaces/junos_l3_interfaces.yml]: ditto

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 176b96ae2f7a9d2dc3b08bcea8018e1a07e35e74

File: models/junos/l3_interfaces/junos_l3_interfaces.yml (hunk 6 above)
Comment (samccann) [reply models/junos/l3_interfaces/junos_l3_interfaces.yml]: thanks!

**Pull request merged**